pkg/gateway/client/session.go

```diff
@@ -81,7 +81,22 @@ func (c *Client) deleteSessionsForUser(ctx context.Context, db *gorm.DB, storage
 	return errors.Join(errs...)
 }
 
+func (c *Client) tableExists(db *gorm.DB, tableName string) (bool, error) {
+	var exists bool
+	if err := db.Raw("SELECT EXISTS (SELECT FROM information_schema.tables WHERE table_name = ?)", tableName).Scan(&exists).Error; err != nil {
+		return false, err
+	}
+	return exists, nil
```

Comment:
nit: Does something like this work?
```suggestion
	return db.Migrator().TableExists(tableName)
```